EKF/ekf.cpp

```diff
@@ -274,29 +254,11 @@ bool Ekf::initialiseFilter()
 		initialiseCovariance();
 
 		// update the yaw angle variance using the variance of the measurement
-		if (_control_status.flags.ev_yaw) {
-			// using error estimate from external vision data TODO: this is never true
-			increaseQuatYawErrVariance(sq(fmaxf(_ev_sample_delayed.angErr, 1.0e-2f)));
-		} else if (_params.mag_fusion_type <= MAG_FUSE_TYPE_3D) {
+		if (_params.mag_fusion_type <= MAG_FUSE_TYPE_3D) {
 			// using magnetic heading tuning parameter
 			increaseQuatYawErrVariance(sq(fmaxf(_params.mag_heading_noise, 1.0e-2f)));
```

Comment:
I think you can also get rid of this one as `increaseQuatYawErrVariance(sq(fmaxf(_params.mag_heading_noise, 1.0e-2f)));` is already called in `resetMagHeading` https://github.com/PX4/ecl/blob/181303488cfc4d2ae34f61f0ad9a34b49a2501f7/EKF/ekf_helper.cpp#L735-L737
after the tilt has been aligned: https://github.com/PX4/ecl/blob/181303488cfc4d2ae34f61f0ad9a34b49a2501f7/EKF/control.cpp#L60

Reply:
Good point. Probably this can be removed as well. But this could be a fundamental functional change. Lets merge this as is and I will open a separate PR for this.